forge/forge/components/code_executor/code_executor.py

```diff
@@ -56,34 +53,38 @@ class CodeExecutionError(CommandExecutionError):
     """The operation (an attempt to run arbitrary code) returned an error"""
 
 
-class CodeExecutorComponent(CommandProvider):
+class CodeExecutorConfiguration(BaseModel):
+    execute_local_commands: bool = False
+    """Enable shell command execution"""
+    shell_command_control: Literal["allowlist"] | Literal["denylist"] = "allowlist"
+    """Controls which list is used"""
+    shell_allowlist: list[str] = Field(default_factory=list)
+    """List of allowed shell commands"""
+    shell_denylist: list[str] = Field(default_factory=list)
+    """List of prohibited shell commands"""
+    docker_container_name: str = "agent_sandbox"
+    """Name of the Docker container used for code execution"""
+
+
+class CodeExecutorComponent(
+    CommandProvider, ConfigurableComponent[CodeExecutorConfiguration]
+):
     """Provides commands to execute Python code and shell commands."""
 
     def __init__(
-        self, workspace: FileStorage, state: BaseAgentSettings, config: Config
+        self,
+        workspace: FileStorage,
+        config: Optional[CodeExecutorConfiguration] = None,
     ):
+        super().__init__(config or CodeExecutorConfiguration())
         self.workspace = workspace
-        self.state = state
-        self.legacy_config = config
-
-        if not we_are_running_in_a_docker_container() and not is_docker_available():
-            logger.info(
-                "Docker is not available or does not support Linux containers. "
-                "The code execution commands will not be available."
-            )
```

Comment:
Where did this notification go?

Reply:
`/dev/null` :) I'll revert this for now. I have an idea how to display it only once even on server but this is outside of scope for this PR.